### PR TITLE
Summary Statistics - Average Over Timeframe

### PR DESCRIPTION
**Pull request by MillerRidgeway**
Refactor the "average" capability to the builder pattern, using new constants outlined in the previous refactoring.

### CODE REVIEW

File: src/main/java/org/sustain/analytics/RollingAverageStatImpl.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                        Constants.DB.HOST,
+                        Constants.DB.PORT)) // mongodb://lattice-46:27017
+                .config("spark.mongodb.input.database", Constants.DB.NAME) // sustaindb
+                .config("spark.mongodb.input.collection", "covid_county_formatted") // future_heat
+                .getOrCreate();
+
+        JavaSparkContext sparkContext = new JavaSparkContext(sparkSession.sparkContext());
+        ReadConfig readConfig = ReadConfig.create(sparkContext);
+
+        String[] jarPaths = {
```
Comment (MillerRidgeway) [inline src/main/java/org/sustain/analytics/RollingAverageStatImpl.java:114]: I noticed that the LinearRegressionModelImpl doesn't have the libraries injected. Mine wouldn't work without it, not sure what I am missing.

**Comment by menuka94** [comment]
@MillerRidgeway: are you planning to include query/protobuf definitions for this feature in a separate PR?

**Comment by MillerRidgeway** [comment]
> @MillerRidgeway: are you planning to include query/protobuf definitions for this feature in a separate PR?

Yes - whenever Shrideep wants to move forward plugging this functionality into the front end, I will plumb the gRPC route fully. For now, I'm focused on other experimentation. If you'd like to see the full gRPC call in this PR we can discuss that, I just didn't want to introduce a lot of changes if they aren't ready to be used.

**Review by menuka94 — APPROVED** [review]

File: src/main/java/org/sustain/analytics/RollingAverageStatImpl.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        log.info("Filtered by daterange");
+
+        Dataset<Row> meanData = dateFiltered
+                .agg(avg("cases"));
+        log.info("Calculated average");
+
+        meanData.show();
+    }
+
+    public static void main(String[] args){
```
Comment (menuka94) [inline src/main/java/org/sustain/analytics/RollingAverageStatImpl.java:100]: We should make this a test (for consistency) later when we make this feature accessible by aperture.

**Pull request merged**